### DISCUSSION TITLE
Captcha asking too often!

### DISCUSSION THREAD

**Discussion by LazyGuy21No** [AsamK/signal-cli#1377] [Q&A]
I am encountering captcha challenges/rate limitations very often (daily) even though I don't use signal much. Is there a way to avoid that?

**Comment by AsamK** [comment]
The exact rate limits aren't public so i can't say. Maybe your IP is in a range that's untrusted by the Signal servers

**Comment by LazyGuy21No** [comment]
Any guess if it maybe has to do with how actively you use **other clients**? 'cause it only happens to a number that is somewhat reserved for an automated service. My private account that is handled via the same IP is working totally fine with signal-cli.

**Comment by LazyGuy21No** [comment]
also: how often do you guys get the captcha?

**Comment by orazioedoardo** [comment]
I never got a captcha other than on registration and I used to send 10s of messages a day. Other than IP address and message count perhaps they also check how many different users you are sending to.

**Reply by LazyGuy21No** [reply]
I seam to be getting it more often, when writing to new people. Anybody got a similar issue?

**Comment by nomansial** [comment]
Im also getting captcha verification even after sending only 2 messages.

@AsamK  Please assist Im working on very important project Please 🙏🙏🙏

**Comment by jacobowitz** [comment]
Any one here who is using signal cli from code in a more automated fashion and willing to share some experiences regarding rate limits and captchas? I am wondering if its feasible to use this project to send like a few hundred messages a day in individual 1:1 chats. With consent of these users ofc. I did not try it yet, but my expectation kind of is that signal is not meant to be used like that (like you could do with telegram bots).
